index.bs

```diff
@@ -1075,12 +1111,68 @@ To <dfn export for="SendStream" lt="create|creating">create</dfn> a
 1. Let |stream| be a [=new=] {{SendStream}}, with:
     : [=[[InternalStream]]=]
     :: |internalStream|
-1. [=WritableStream/Set up=] |stream| with TBD.
-1. [=set/Add=] |stream| to |transport|'s [=[[SendStreams]]=].
+    : [=[[Transport]]=]
+    :: |transport|
+1. Let |writeAlgorithm| be an action that [=writes=] |chunk| with |stream|, given |chunk|.
+1. Let |closeAlgorithm| be an action that [=closes=] |stream|.
+1. Let |abortAlgorithm| be an action that [=aborts=] |stream| with |reason|, given |reason|.
+1. [=WritableStream/Set up=] |stream| with [=WritableStream/set up/writeAlgorithm=] set to
+   |writeAlgorithm|, [=WritableStream/set up/closeAlgorithm=] set to |closeAlgorithm|,
+   [=WritableStream/set up/abortAlgorithm=] set to |abortAlgorithm|.
+1. [=set/Append=] |stream| to |transport|'s [=[[SendStreams]]=].
 1. Return |stream|.
 
 </div>
 
+<div algorithm>
+To <dfn for="SendStream">write</dfn> |chunk| with a {{SendStream}} |stream|, run these steps:
+
+1. Let |transport| be |stream|'s [=[[Transport]]=].
+1. If |chunk| is not a {{Uint8Array}}, return [=a promise rejected with=] a {{TypeError}}.
+1. Let |promise| be a new promise.
+1. Let |bytes| be a copy of the [=byte sequence=] which |chunk| represents.
+1. Return |promise| and run the remaining steps [=in parallel=].
+1. [=stream/Send=] |bytes| on |stream|'s [=[[InternalStream]]=] and wait for the operation to
+   complete.
+1. [=Queue a network task=] with |transport| to [=resolve=] |promise| with undefined.
+
+Note: The user-agent MAY have a buffer to improve the transfer performance. Such a buffer
+SHOULD have a fixed size, to carry the backpressure information to the user of {{SendStream}}. This
+also means the [=fulfillment=] of the promise returned from this algorithm (or,
+{{WritableStreamDefaultWriter.write}}) does **NOT** necessarily mean that the chunk is acked by
+the server [[!QUIC]]. It may just mean that the chunk is just appended to the buffer. To
+make sure that the chunk arrives at the server, close the {{SendStream}} and wait for the
+result promise to be fulfilled, or use an application-level protocol.
```

Comment:
I don't think we want to encourage relying on the `close()` promise, because even if the chunk arrives at the server, that does not necessarily mean that the server application has processed the data in question.

Reply:
Fixed.